### PR TITLE
Reenable forked mainnet

### PR DESCRIPTION
**Pull request by peterargue**
Closes #???

## Description

Backports: https://github.com/onflow/flow-emulator/pull/594 to master

This PR reenables forked network support and updates it to use Access nodes as the archive backend
______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the GitHub PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by peterargue** [comment]
@turbolent here's the forked mainnet backport

File: server/server.go [hunk 1]
```diff
@@ -345,6 +348,23 @@ func configureStorage(conf *Config) (storageProvider storage.Store, err error) {
 		}
 	}
 
+	if conf.ChainID == flowgo.Testnet || conf.ChainID == flowgo.Mainnet {
+		// TODO: any reason redis shouldn't work?
```
Comment (sideninja) [inline server/server.go:352]: in the current design because the remote impl relies on sqlite instance as the provider, but if refactored I don't think there should be a reason.

File: internal/executiondata.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package internal
+
+import (
+	"github.com/onflow/flow/protobuf/go/flow/executiondata"
+)
+
+type ExecutionDataAPIClient executiondata.ExecutionDataAPIClient
```
Comment (sideninja) [inline internal/executiondata.go:25]: a comment why is this needed

**Review by devbugging — APPROVED** [review]
nice

File: internal/executiondata.go (hunk 2 above)
Comment (peterargue) [reply internal/executiondata.go:25]: it's actually not in this PR. it was needed for the tests that I left out
